mlflow/R/mlflow/R/tracking-experiments.R

```diff
@@ -182,18 +181,18 @@ mlflow_set_experiment <- function(experiment_name = NULL, experiment_id = NULL,
 
   client <- mlflow_client()
 
-  experiment <- if (!is.null(experiment_name)) {
+  final_experiment_id <- if (!is.null(experiment_name)) {
     tryCatch(
-      mlflow_get_experiment(client = client, name = experiment_name),
+      mlflow_id(mlflow_get_experiment(client = client, name = experiment_name)),
       error = function(e) {
         message("Experiment `", experiment_name, "` does not exist. Creating a new experiment.")
         mlflow_create_experiment(client = client, name = experiment_name, artifact_location = artifact_location)
       }
     )
   } else {
-    mlflow_get_experiment(client = client, experiment_id = experiment_id)
+    experiment_id
   }
 
-  mlflow_set_active_experiment_id(mlflow_id(experiment))
-  experiment
+  mlflow_set_active_experiment_id(final_experiment_id)
+  NULL
```

Comment:
we could return invisible(final_experiment_id)? do we not do it to be consistent with python? I think we already do not have direct equivalent, do we?

Reply:
Yep we could, seems ok to return from `mlflow_set_experiment` in this case, especially since there is no exact Python equivalent to this API (it accepts name or experiment ID whereas the Python `set_experiment` API only takes name

Reply:
Changed this to return